client/components/download-header.js

```diff
@@ -1,14 +1,64 @@
-import React from 'react';
+import React, { useState, useRef, useEffect, Fragment } from 'react';
+import { connect } from 'react-redux';
+import DownloadLink from './download-link';
 
-export default ({ title }) => (
-  <div className="download-header">
-    <div className="govuk-grid-row">
-      <div className="govuk-grid-column-three-quarters">
-        <h3>{title}</h3>
+function DownloadHeader({ title, isGranted, drafting, id, basename }) {
+  const [modalShowing, updateModalShowing] = useState(false);
+  const container = useRef(null);
+  const download = useRef(null);
+  basename = basename.replace(/\/edit/, '');
+
+  // project title could span multiple lines, adjust download position accordingly
+  useEffect(() => {
+    // subtract padding, and border
+    const height = container.current.offsetHeight - 30 - 4;
+    download.current.style.height = `${height}px`;
+    download.current.style.lineHeight = `${height}px`;
```

Comment:
Is this setting inline styles? Does that actually work with CSP policies as they are?

Reply:
Spoiler: it shouldn't.

Reply:
it doesn't, updating the dom element in this way is fine with CSP, it's only if you set inline styles on the element using the style attribute

Reply:
also this is client side, so CSP has already done it's shizzle by the time this runs

Reply:
I'm not sure that last bit is true - CSP should prevent new elements which violate the policy being added, but if it works then I'm happy.

Reply:
ORLY? I did not know that. However this does work